internal/beater/config/rum.go

```diff
@@ -72,20 +74,35 @@ func (c *RumConfig) setup(log *logp.Logger, outputESCfg *config.C) error {
 		return errors.Wrapf(err, "Invalid regex for `exclude_from_grouping`: ")
 	}
 
-	// No need to unpack the ESConfig if SourceMapMetadata exist
 	if len(c.SourceMapping.Metadata) > 0 {
-		return nil
+		// We don't have the fleet fetcher anymore.
+		// Ignore metadata and setup the elasticsearch config.
+		log.Warn("Ignoring sourcemap metadata")
```

Comment:
I'd completely remove `SourceMapping.Metadata`. Afaics this is not used anywhere anymore and was only used with the FleetFetcher, so it should never reach this code.

Reply:
It is not used in the APM Server but I think kibana (?) might be injecting this option. 

For example, in  `TestRUMErrorSourcemapping` this code path is triggered and we're logging a warning.

I agree we can remove it here :+1: